app/src/components/guestbook/GuestBookMessageEdit.tsx

```diff
@@ -34,45 +41,93 @@ export default function GuestBookMessageEdit({
-                ...newValues,
-            });
-        },
-        [message, setMessage]
-    );
+
+    const initializeValues = (): NewGuestBookMessage | IGuestBookMessage => {
+        if (isUndefined(message)){
+            return getInitialMessage()
+        }
+        return message!
+    }
+
+    const validationSchema = Yup.object({
+        author: Yup.string()
+            .matches(
+                /^([A-Za-z\u00C0-\u00D6\u00D8-\u00f6\u00f8-\u00ff\s]*)$/gi,
+                'Name can only contain Latin letters.'
+            )
+            .min(2, 'Too short!')
+            .max(50, 'Too long!')
+            .required('Required!'),
+        message: Yup.string().min(2, 'Too short!').max(1000, 'Too long!').required('Required!'),
+    });
 
     return (
-        <Form>
-            <Form.Group className="mb-3" controlId="EditMessage.Author">
-                <Form.Label>Name</Form.Label>
-                <Form.Control
-                    type="text"
-                    placeholder={placeholderName}
-                    defaultValue={message?.author}
-                    onChange={(event) => updateMessage({ author: event.target.value })}
-                />
-            </Form.Group>
-            <Form.Group className="mb-3" controlId="EditMessage.Message">
-                <Form.Label>Message</Form.Label>
-                <Form.Control
-                    as="textarea"
-                    rows={3}
-                    defaultValue={message?.message}
-                    onChange={(event) => updateMessage({ message: event.target.value })}
-                />
-            </Form.Group>
-            <Stack direction="horizontal" gap={3} className="justify-content-end">
-                <div className="my-auto"></div>
-                <ButtonLink variant="secondary" href={cancelHref}>
-                    Cancel
-                </ButtonLink>
-                <Button variant="primary" onClick={onSubmit} disabled={isSubmitLoading}>
-                    {isSubmitLoading ? <Spinner animation="border" size="sm" /> : submitLabel}
-                </Button>
-            </Stack>
-        </Form>
+        <Formik
+            initialValues={initializeValues()}
+            validationSchema={validationSchema}
+            onSubmit={(message, { setSubmitting }) => {
+                try {
+                    if (onEdit && 'messageId' in message){
+                        console.log('edit new message')
+                        onEdit(message as IGuestBookMessage)
+                    } else if (onCreate) {
+                        console.log('create a message')
+                        onCreate(message as NewGuestBookMessage)
+                    }
+                } catch (error) {
+                    console.log('Failed to create message', error);
+                } finally {
+                    setSubmitting(false);
+                }
+            }}
+        >
+            {(props) => (
```

Comment:
Is this part necessary? I think you're creating an anonymous subcomponent which is not needed. The structure can be simplified to
```
<Formik ...>
  <Form ...>
    ...
  </Form>
</Formik>
```

Reply:
If I understand correctly, you want to simplify it by removing 'props' variable ? 
The 'props' variable is used to access Formik's functions like handleSubmit and Formik's variables like values (values of the field being track).
I can do something like this, but I'm not sure if it's better 

`<Formik
  ....
  {({ handleSubmit, handleBlur, handleChange, values }) => (
    <Form onSubmit={handleSubmit}>
      <Form.Group className="mb-3" controlId="EditMessage.Author">
        <Form.Label>Name</Form.Label>
        <Form.Control
          name="author"
          type="text"
          placeholder={placeholderName}
          value={values.author}
          onBlur={handleBlur}
          onChange={handleChange}
        />
  )}
/>
`